include/swift/AST/TypeCheckRequests.h

```diff
@@ -2141,6 +2141,25 @@ class ScopedImportLookupRequest
   bool isCached() const { return true; }
 };
 
+/// Determines whether this is a "simple" didSet i.e one that either does not
+/// use the implicit oldValue parameter in the body or does not take an explicit
+/// parameter (ex: 'didSet(oldValue)').
+class SimpleDidSetRequest
+    : public SimpleRequest<SimpleDidSetRequest, bool(AccessorDecl *),
+                           CacheKind::Cached> {
+public:
+  using SimpleRequest::SimpleRequest;
+
+private:
+  friend SimpleRequest;
+
+  // Evaluation.
+  llvm::Expected<bool> evaluate(Evaluator &evaluator, AccessorDecl *decl) const;
+
+public:
+  bool isCached() const { return true; }
```

Comment:
Based on my comment below, I'd like this request to be evaluated for all accessors. However, there's no need to cache for non-didSet accessors. Can you change the implementation of `isCached()` to:

```
return std::get<0>(getStorage())->getAccessorKind() == AccessorKind::DidSet;
```
?